### PR TITLE
[libhal] Add library to conan center index

### PR DESCRIPTION
**Pull request by kammce**
Specify library name and version:  **libhal/0.0.0**

New addition to ConanCenter.
I am the author of the project.
Will be the dependency of future libraries.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@kammce thank you for your contribution, please, don't forget to sign the CLA on this PR and request your access on the issue #4. After that, the CI bot should run your recipe. In case of doubt of the review process, please, read https://github.com/conan-io/conan-center-index/blob/master/docs/review_process.md. For better recipe quality, you can use this guide: https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md

**Review by uilianries — COMMENTED** [review]
@libhal consider https://github.com/libhal/conan-center-index/pull/1

**Comment by kammce** [comment]
@uilianries I added myself to the #4 a while back. When can I expect access?

**Comment by kammce** [comment]
> @libhal consider [libhal#1](https://github.com/libhal/conan-center-index/pull/1)

Thank you! I will!

**Comment by kammce** [comment]
@uilianries when I run `conan create . --version=0.0.0` using conan I get:

```
ERROR: libhal/0.0.0: Error in validate() method, line 45
        if self.settings.compiler.get_safe("cppstd"):
        ConanException: 'self.settings' access in 'validate()' method is forbidden
```

Whats the correct way to get the settings for conan v2? 
My version is `Conan version 2.0.0-beta2`

**Comment by uilianries** [comment]
@kammce You need to run conan 1.51.3

   pip install -U conan

or 

   pip install conan==1.51.3

**Comment by kammce** [comment]
> @kammce You need to run conan 1.51.3
> 
> pip install -U conan
> 
> or
> 
> pip install conan==1.51.3

Oh yes, it works with 1.51.3, but I thought the point of preparing for 2.0 was such that 2.0 would work? Or should we ignore running this on conan 2.0 because it is still in beta?

**Comment by kammce** [comment]
Anything else I need or am I just waiting on approval?

**Comment by uilianries** [comment]
@kammce The idea is be compatible for both. To access self.settings in validate(), you should use `self.info.settings` same for options. More info: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-validate-method

**Comment by uilianries** [comment]
> @uilianries self.info.settings doesn't work for me on Conan 1.51.3

Does not work how? It's supported since previous versions. Do you have a log? Anyway, you can keep `self.settings` only for now.

**Comment by kammce** [comment]
@uilianries 

```
$ conan --version
Conan version 1.51.3
```

I get this error:

```
ERROR: libhal/0.0.0: Error in validate() method, line 41
        if self.info.settings.compiler.get_safe("cppstd"):
        AttributeError: 'NoneType' object has no attribute 'get_safe'
```
if I comment that section out I get this

```
ERROR: libhal/0.0.0: Error in validate() method, line 50
        version = str(self.info.settings.compiler.version)
        AttributeError: 'NoneType' object has no attribute 'version'
```

printing `self.info` results in `<conans.model.info.ConanInfo object at 0x102b43310>`

```
ERROR: libhal/0.0.0: Error in validate() method, line 41
        print(self.info.compiler)
        AttributeError: 'ConanInfo' object has no attribute 'compiler'
```

**Comment by uilianries** [comment]
@kammce It's `self.settings.get_safe("compiler.cppstd")`

**Review by uilianries — APPROVED** [review]
LGTM!

**Comment by kammce** [comment]
WOOT! Looks like I need two more reviewers? How do I get more reviewers?

File: recipes/libhal/all/conanfile.py
Comment (SpaceIm) [inline recipes/libhal/all/conanfile.py:39]:
```suggestion
            "msvc": "193",
```

File: recipes/libhal/all/conanfile.py
Comment (SpaceIm) [inline recipes/libhal/all/conanfile.py:62]:
```suggestion
        if minimum_version and lazy_lt_semver(version, minimum_version):
            raise ConanInvalidConfiguration(
                f"{self.name} {self.version} requires C++{self._min_cppstd}, which your compiler ({compiler}-{version}) does not support")
```

Comment (SpaceIm) [reply recipes/libhal/all/conanfile.py:62]: don't raise when you don't know the compiler.

File: recipes/libhal/all/conanfile.py
Comment (SpaceIm) [inline recipes/libhal/all/conanfile.py:29]:
```suggestion
```

File: recipes/libhal/all/conanfile.py
Comment (SpaceIm) [inline recipes/libhal/all/conanfile.py:66]:
```suggestion

    def source(self):
        get(self, **self.conan_data["sources"][self.version],
            destination=self.source_folder, strip_root=True)

```

source() here to understand the flow of execution when reading the recipe

File: recipes/libhal/all/conanfile.py
Comment (SpaceIm) [inline recipes/libhal/all/conanfile.py:73]:
```suggestion
             "include"), src=os.path.join(self.source_folder, "include"))
 
    def package_info(self):
        self.cpp_info.bindirs = []
        self.cpp_info.frameworkdirs = []
        self.cpp_info.libdirs = []
        self.cpp_info.resdirs = []
```

File: recipes/libhal/all/test_package/CMakeLists.txt
Comment (SpaceIm) [inline recipes/libhal/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.8)
```

File: recipes/libhal/all/test_v1_package/CMakeLists.txt
Comment (SpaceIm) [inline recipes/libhal/all/test_v1_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.8)
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**
